diff --git a/pkg/daemons/control/server.go b/pkg/daemons/control/server.go
index 41a4820691..19ad80d074 100644
--- a/pkg/daemons/control/server.go
+++ b/pkg/daemons/control/server.go
@@ -3,8 +3,10 @@
 import (
 	"context"
 	"errors"
+	"log"
 	"os"
 	"path/filepath"
+	"slices"
 	"strconv"
 	"strings"
 	"sync"
@@ -278,6 +280,19 @@
 
 	args := util.GetArgs(argsMap, cfg.ExtraAPIArgs)
 
+	log.Printf("++ args before: %#v", args)
+
+	if slices.ContainsFunc(args, func(s string) bool {
+		return strings.HasPrefix(s, "--service-account-signing-endpoint=")
+	}) {
+		args = slices.DeleteFunc(args, func(s string) bool {
+			return strings.HasPrefix(s, "--service-account-signing-key-file=") ||
+				strings.HasPrefix(s, "--service-account-key-file=")
+		})
+	}
+
+	log.Printf("++ args after: %#v", args)
+
 	logrus.Infof("Running kube-apiserver %s", config.ArgString(args))
 
 	return executor.APIServer(ctx, args)
